server/core/src/main/java/io/whitefox/core/MetastoreProperties.java

```diff
@@ -51,4 +51,41 @@ public String toString() {
           + credentials + ']';
     }
   }
+
+  final class HadoopMetastoreProperties implements MetastoreProperties {
+    private final String location;
+
+    public HadoopMetastoreProperties(String location, MetastoreType type) {
+      if (type != MetastoreType.HADOOP) {
+        throw new IllegalArgumentException(String.format(
+            "GlueMetastore properties are not compatible with metastore of type %o", type));
```

Comment:
```suggestion
            "Hadoop metastore properties are not compatible with metastore of type %o", type));
```

Reply:
you will never find those tests because I am a good programmer and I keep the pr small. In this task we want to implement a table loader for iceberg and nothing more. In the next issue implement the read API (getMetadata, getTableVersion and queryTable) and finally, we will have another issue to test those api with a spark client. 😄 

I've created the issues for the implementation of all read apis.